### PR TITLE
⬆️ Upgraded dependencies to allow PHP >= 8.0 and Laravel 8.0

### PR DESCRIPTION
**Pull request by yoan-myparcel**
# Changes
- Added support for PHP 8.0
- Added support for Laravel 8.0
- Updated phpunit version + new formatted config file
- Updated to composer 2.0
- Added proper user permissions for docker-compose
- Unmocked Iterator :skull_and_crossbones:

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -4,11 +4,11 @@
   "homepage": "https://www.myparcel.com",
   "type": "library",
   "require": {
-    "php": ">=7.1",
+    "php": "^7.1|^8.0",
```
Comment (M4tini) [inline composer.json:7]: I believe this is not necessary, unless the goal is to exclude ^9.0

File: composer.json [hunk 2]
```diff
@@ -4,11 +4,11 @@
...
     "ext-json": "*",
     "guzzlehttp/promises": "^1.0",
     "justinrainbow/json-schema": "^5.2",
-    "laravel/framework": "^5.4|^6.0|^7.0"
+    "laravel/framework": "^5.4|^6.0|^7.0|^8.0"
```
Comment (M4tini) [inline composer.json:11]: If we go the `>=` route above, it makes sense to do it here as well.
```suggestion
    "laravel/framework": ">=5.4"
```

File: docker-compose.yml [hunk 3]
```diff
@@ -5,11 +5,12 @@ services:
     build:
       context: ./docker
       dockerfile: Dockerfile
-    image: myparcelcom/php:7.3-cli
+    image: myparcelcom/json-api:v1-dev
     working_dir: /opt/json-api
     volumes:
       - .:/opt/json-api
       - composercache:/tmp
+    user: ${CURRENT_UID}
```
Comment (M4tini) [inline docker-compose.yml:13]: Just wondering, can we do the same with the auth server where we have permission issues with the .key files?

**Review by M4tini — DISMISSED** [review]

File: docker-compose.yml (hunk 3 above)
Comment (NickVries) [reply docker-compose.yml:13]: Yes! We can actually do this in every project! This makes the container run as the current user (you) and we no longer have to fix the permissions using those shell functions in the `mp.sh` script!
I got this idea from Sam over the weekend!

**Comment by yoan-myparcel** [comment]
@M4tini I used your suggestions for the dependency versions. Can you review again? Thanks!

**Review by M4tini — APPROVED** [review]

**Pull request merged**